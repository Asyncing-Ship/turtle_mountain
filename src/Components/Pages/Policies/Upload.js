import React from "react";
import { Button, useToast } from "@chakra-ui/core";
import { connect } from "react-redux";
import * as filestack from "filestack-js";

const client = filestack.init(`${process.env.REACT_APP_FILESTACK_KEY}`);

const Upload = (props) => {
  const toast = useToast();

  const uploadFiles = async (data) => {
    await props.dispatch({
      type: "UPLOAD_POLICY",
      payload: { filename: data.filename, handle: data.handle },
    });
  };

  const options = {
    fromSources: ["local_file_system", "googledrive", "facebook"],
    onFileUploadFinished: (res) => {
      uploadFiles({
        filename: res.filename,
        handle: res.handle,
      }).then((res) => {
        toast({
          title: "Uploaded!",
          description: "You successfully uploaded a file.",
          status: "info",
          duration: 5000,
          isClosable: true,
          position: "bottom-right",
        });
      });
    },
    onUploadDone: () => {
      toast({
        title: "Done!",
        description: "All files have finished uploading.",
        status: "success",
        duration: 5000,
        isClosable: true,
        position: "bottom-right",
      });
    },
    onFileUploadFailed: (res) => {
      toast({
        title: "Failed!",
        description: `${res.filename} failed to upload.`,
        status: "warning",
        duration: 5000,
        isClosable: true,
        position: "bottom-right",
      });
    },
    minFiles: 1,
    maxFiles: 5,
    maxSize: 1 * 1024 * 1024, // 1 MB file size limit
  };

  return (
    <>
      <Button
        onClick={async () => {
          await client.picker(options).open();
        }}
      >
        Upload Policy
      </Button>
    </>
  );
};

const mapStateToProps = (state) => {
  return { user: state.user };
};

export default connect(mapStateToProps)(Upload);
