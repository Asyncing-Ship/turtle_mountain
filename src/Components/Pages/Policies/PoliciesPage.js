import React, { Component } from 'react';
import Upload from './Upload';
import { connect } from 'react-redux';
import { SimpleGrid, Box, Stack, Text, Button, ButtonGroup } from '@chakra-ui/core';
import PolicyModal from './PolicyModal';

class PoliciesPage extends Component {
  componentDidMount() {
    this.props.dispatch({ type: 'FETCH_POLICIES' });
  }

  render() {
    const { policies } = this.props;
    return (
      <>
        <Stack>
          <Box p={5}>
            <Upload />
          </Box>
          <SimpleGrid columns={{ sm: 2, md: 3, lg: 4 }}>
            {
              policies.map((x, i) =>
                <Box
                  color="#f5fffa"
                  rounded="lg"
                  p={3}
                  m={3}
                  bg="#2f2e2e"
                  key={i}>
                  <Stack>
                    <Text>{x.filename}</Text>
                    <ButtonGroup>
                      {/* <Link
                        target="_blank"
                        rel="noopener noreferrer"
                        href={`https://cdn.filestackcontent.com/${x.handle}`}
                      >
                        <Button rightIcon="view" size="sm" variantColor="blue">View</Button>
                      </Link> */}
                      <PolicyModal x={x} />
                      <Button rightIcon="download" size="sm" variantColor="purple">
                        <a
                          target="_blank"
                          rel="noopener noreferrer"
                          href={`https://cdn.filestackcontent.com/${x.handle}`}
                          download={x.filename}
                        >
                          Download
                        </a>
                      </Button>
                    </ButtonGroup>
                  </Stack>
                </Box>
              )
            }
          </SimpleGrid>
        </Stack>
      </>
    )
  }
}

const mapStateToProps = state => {
  return {
    policies: state.policies,
  }
}

export default connect(mapStateToProps)(PoliciesPage);