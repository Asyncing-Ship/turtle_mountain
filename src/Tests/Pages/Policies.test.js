import React from "react";
import { shallow } from "enzyme";
import PoliciesPage from "../../Components/Pages/Policies/PoliciesPage";
import Upload from "../../Components/Pages/Policies/Upload";
import toJson from "enzyme-to-json";
import { createStore } from "redux";
import rootReducer from "../../Redux/Reducers/index";
it("renders without crashing", () => {
  shallow(<PoliciesPage />);
});
it("renders without crashing", () => {
  shallow(<Upload />);
});
