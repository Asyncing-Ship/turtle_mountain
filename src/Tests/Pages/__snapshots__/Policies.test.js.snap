// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Fragment>
  This is the policy page!
</Fragment>
`;

exports[`renders without crashing 2`] = `
<Fragment>
  Upload a new policy here:
</Fragment>
`;
